unstructured/ingest/connector/sharepoint.py

```diff
@@ -373,7 +373,18 @@ def initialize(self):
     def get_ingest_docs(self):
         base_site_client = self.connector_config.get_site_client()
 
-        if self.connector_config.permissions_config:
+        if not all(
+            [  # noqa: C419
+                value
+                for key, value in self.connector_config.permissions_config.to_dict().items()
+                if key in ["application_id", "client_cred", "tenant"]
```

Comment:
can't we just do `getattr(self.connector_config.permissions_config, attr) for attr in ["application_id", "client_cred", "tenant"]`

Reply:
that way we avoid running extra dict conversion

Reply:
Agreed, added with https://github.com/Unstructured-IO/unstructured/pull/1894/commits/0dc1a981854be4e69ee76e0e850c090a37ab49ed